src/dotless.Core/Parser/Tree/Ruleset.cs

```diff
@@ -279,6 +281,50 @@ protected void EvaluateRules(Env env)
             env.Frames.Pop();
         }
 
+        /// <summary>
+        /// Merges properties with merge operators '+' and '+_'
+        /// </summary>
+        /// <returns>New merged ruleset</returns>
+        private NodeList MergeProperties()
+        {
+            var groups = new Dictionary<string, List<Rule>>();
+            var sorted = new List<Node>();
+            foreach (var node in Rules)
+            {
+                var rule = node as Rule;
+                if (rule == null || rule.Merge == "")
+                {
+                    sorted.Add(node);
+                    continue;
+                }
+                var key = $"{rule.Name},{rule.Important},{rule.Merge}";
+                if (!groups.ContainsKey(key))
+                {
+                    groups[key] = new List<Rule> {rule};
+                    sorted.Add(node);
```

Comment:
Order needs to be preserved since multiple rules for the same property must me merged or overwritten in the correct order for the selector.

Ex.

`
@paddingTopBottom:10px;
@paddingRightLeft:20px;
.padding(num)  { padding+_: num; }
div {
    .padding(@paddingTopBottom);
    .padding(@paddingRightLeft);
}`
(In this case the two padding rules might also come through more convoluted paths so it might be as obvious as this.)
 
This, after expands, ends up:
`
div {
    .padding+_:10px;
    .padding+_20px;
}`

And is merged to
`
    div { padding: 10px 20px; }`

If they get out of order the padding will be switched.

But I can look at the code and see if I can make it simpler or at least explain what is done better.

The first element in each group are added to the sorted list directly and when I then merge the group the element is already in the correct position in the resulting set of rules.
Any following merge rules for the same selector is only added to the merge group, not to the sorted output.